src/meta/src/lib.rs

```diff
@@ -46,114 +46,50 @@ mod rpc;
 use std::time::Duration;
 
-use clap::{ArgEnum, Parser};
 pub use error::{MetaError, MetaResult};
+use risingwave_common::config::{load_config, MetaBackend, MetaConfig};
 
 use crate::manager::MetaOpts;
 use crate::rpc::server::{rpc_serve, AddressInfo, MetaStoreBackend};
 
-#[derive(Copy, Clone, Debug, ArgEnum)]
-pub enum Backend {
-    Mem,
-    Etcd,
-}
-
-#[derive(Debug, Clone, Parser)]
-pub struct MetaNodeOpts {
-    // TODO: rename to listen_address and separate out the port.
-    #[clap(long, default_value = "127.0.0.1:5690")]
-    listen_addr: String,
-
-    /// The endpoint for this meta node, which also serves as its unique identifier in cluster
-    /// membership and leader election.
-    #[clap(long)]
-    meta_endpoint: Option<String>,
-
-    #[clap(long)]
-    dashboard_host: Option<String>,
-
-    #[clap(long)]
-    prometheus_host: Option<String>,
-
-    #[clap(long, arg_enum, default_value_t = Backend::Mem)]
-    backend: Backend,
-
-    #[clap(long, default_value_t = String::from(""))]
-    etcd_endpoints: String,
-
-    /// Enable authentication with etcd. By default disabled.
-    #[clap(long)]
-    etcd_auth: bool,
-
-    /// Username of etcd, required when --etcd-auth is enabled.
-    /// Default value is read from the 'ETCD_USERNAME' environment variable.
-    #[clap(long, env = "ETCD_USERNAME", default_value = "")]
-    etcd_username: String,
-
-    /// Password of etcd, required when --etcd-auth is enabled.
-    /// Default value is read from the 'ETCD_PASSWORD' environment variable.
-    #[clap(long, env = "ETCD_PASSWORD", default_value = "")]
-    etcd_password: String,
-
-    #[clap(long)]
-    dashboard_ui_path: Option<String>,
-
-    /// For dashboard service to fetch cluster info.
-    #[clap(long)]
-    prometheus_endpoint: Option<String>,
-
-    /// Endpoint of the connector node, there will be a sidecar connector node
-    /// colocated with Meta node in the cloud environment
-    #[clap(long, env = "META_CONNECTOR_RPC_ENDPOINT")]
-    pub connector_rpc_endpoint: Option<String>,
-
-    /// The path of `risingwave.toml` configuration file.
-    ///
-    /// If empty, default configuration values will be used.
-    ///
-    /// Note that internal system parameters should be defined in the configuration file at
-    /// [`risingwave_common::config`] instead of command line arguments.
-    #[clap(long, default_value = "")]
-    pub config_path: String,
-}
-
-use std::future::Future;
-use std::pin::Pin;
-
-use risingwave_common::config::load_config;
-
 /// Start meta node
-pub fn start(opts: MetaNodeOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+pub fn start(opts: MetaConfig) -> Pin<Box<dyn Future<Output = ()> + Send>> {
     // WARNING: don't change the function signature. Making it `async fn` will cause
     // slow compile in release mode.
     Box::pin(async move {
-        let config = load_config(&opts.config_path);
+        let config = load_config(&opts.config_path.clone(), Some(opts));
```

Comment:
Is `opts` only used for CLI overwrite here? If so, it'll be better to give it a more specific name to show that this should not be used as the final config. Same for other occurrences.

Reply:
`load_config` takes ownership of `opts`, so it should not be usable after the invocation 🤔

Reply:
Although it's possible that someone may want to clone it 😅 Maybe some more comments?